test/test.cmake

```diff
@@ -1,14 +1,28 @@
 include(CTest)
 
-add_library(test-lib ${CMAKE_CURRENT_SOURCE_DIR}/../test-lib.c)
+add_library(test-lib-mesen ${CMAKE_CURRENT_SOURCE_DIR}/../test-lib-mesen.c)
+target_include_directories(test-lib-mesen PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../)
+
+add_library(test-lib-emutest ${CMAKE_CURRENT_SOURCE_DIR}/../test-lib-emutest.c)
+target_include_directories(test-lib-emutest PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../)
 
 function(add_mesen_test name)
   add_executable(${name} ${name}.c)
-  target_link_libraries(${name} test-lib)
+  target_link_libraries(${name} test-lib-mesen)
   add_test(NAME test-${name} COMMAND ${MESEN_COMMAND} --testrunner
            $<TARGET_FILE:${name}> ${CMAKE_CURRENT_SOURCE_DIR}/../mesen.lua)
 endfunction()
 
+function(add_emutest_test name binext source_dir libretro_core)
```

Comment:
Before, the test suite could (for the most part) only be run if mesen was available, so there was little point in trying to without it installed. Now that the dependency list is growing, it should be possible to partially run the test suite with only some of the dependencies installed. This makes it possible to test and fix CI breakages locally without needing to install the whole CI world (which will become increasingly intractible).

Reply:
Yep, CMake finds all Libretro cores and will fail tests where you don't have them installed. You can run e.g. "ninja test-nes-nrom" for example to run a single test project. I could add a short section to the README with links to dependencies and instructions.